velox/type/tests/FilterTest.cpp

```diff
@@ -1729,3 +1729,43 @@ TEST(FilterTest, mergeWithBytesMultiRange) {
     }
   }
 }
+
+TEST(FilterTest, HugeIntRange) {
+  int128_t input = 1;
+  auto filter = equalHugeint(input, false);
+  auto testInt128 = [&](int128_t x) { return filter->testInt128(x); };
+  EXPECT_TRUE(filter->testInt128(1));
+
+  EXPECT_FALSE(filter->testNull());
+  EXPECT_FALSE(filter->testInt128(0));
+
+  EXPECT_FALSE(filter->testInt128Range(100, 150, false));
+  EXPECT_FALSE(filter->testInt128Range(100, 150, true));
+  EXPECT_TRUE(filter->testInt128Range(1, 10, false));
+  EXPECT_TRUE(filter->testInt128Range(-10, 10, false));
+
+  // x between 1 and 10
+  filter = betweenHugeint(1, 10, false);
+  EXPECT_TRUE(filter->testInt128(1));
+  EXPECT_TRUE(filter->testInt128(10));
+
+  EXPECT_FALSE(filter->testNull());
+  EXPECT_FALSE(filter->testInt128(0));
+  EXPECT_FALSE(filter->testInt128(11));
+
+  EXPECT_FALSE(filter->testInt128Range(-150, -10, false));
+  EXPECT_FALSE(filter->testInt128Range(-150, -10, true));
+  EXPECT_TRUE(filter->testInt128Range(5, 20, false));
+  EXPECT_TRUE(filter->testInt128Range(-10, 10, false));
+
+  // x > 0 or null
+  filter = greaterThanHugeint(0, true);
```

Comment:
Its a bit odd to test bigDecimal with values like 0. Please add testing with values that are more representative of BigDecimal. Also add tests for boundary conditions.

Reply:
Updated, thanks for the suggestion.